docs/chart_template_guide/named_templates.md

```diff
@@ -4,15 +4,19 @@ It is time to move beyond one template, and begin to create others. In this sect
 
 In the "Flow Control" section we introduced three actions for declaring and managing templates: `define`, `template`, and `block`. In this section, we'll cover those three actions, and also introduce a special-purpose `include` function that works similarly to the `template` action.
 
+An important detail to keep in mind when naming templates: **template names are global**. If you declare two templates with the same name, whichever one is loaded last will be the one used. Because templates in subcharts are compiled together with top-level templates, you should be careful to name your templates with _chart-specific names_.
+
+One popular naming convention is to prefix each defined template with the name of the chart: `{{ define "mychart.labels" }}` or `{{ define "mychart_labels" }}`. By using the specific chart name as a prefix we can avoid any conflicts that may arise due to two different charts that implement templates of the same name.
```

Comment:
Please use `{{ define "mychart.labels" }}` only. This is also what `helm create` does and what we expect as best practice in the charts repo.